File: storage/db-core/src/main/java/com/coffee/gu/coupon/CouponTargetEntity.java
package com.coffee.gu.coupon;

import com.coffee.gu.enums.CouponTargetType;
import com.coffee.gu.BaseEntity;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.Table;

@Table(name = "coupon_target")
@Entity
public class CouponTargetEntity extends BaseEntity {
    private Long couponId;
    @Enumerated(EnumType.STRING)
    private CouponTargetType targetType;
    private Long targetId;

    protected CouponTargetEntity() {}

    public CouponTargetEntity(Long couponId, CouponTargetType targetType, Long targetId) {
        this.couponId = couponId;
        this.targetType = targetType;
        this.targetId = targetId;
    }

    public CouponTarget toModel() {
        return new CouponTarget(id, couponId, targetType, targetId);
    }

    public Long getCouponId() {
        return couponId;
    }

    public CouponTargetType getTargetType() {
        return targetType;
    }

    public Long getTargetId() {
        return targetId;
    }
}

File: storage/db-core/src/main/kotlin/com/coffee/gu/coupon/CouponTargetEntity.kt
package com.coffee.gu.coupon

import com.coffee.gu.BaseEntity
import com.coffee.gu.enums.CouponTargetType
import jakarta.persistence.Entity
import jakarta.persistence.EnumType
import jakarta.persistence.Enumerated
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id
import jakarta.persistence.Table

@Table(name = "coupon_target")
@Entity
class CouponTargetEntity @JvmOverloads constructor(
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long = 0,
    val couponId: Long,
    @Enumerated(EnumType.STRING)
    val targetType: CouponTargetType,
    val targetId: Long,
) : BaseEntity() {

    fun toModel(): CouponTarget {
        return CouponTarget(
            id = id,
            couponId = couponId,
            targetType = targetType,
            targetId = targetId,
        )
    }
}
